tests/OAuth2ClientHandler.Tests/Helpers/Startup.cs

```diff
@@ -22,7 +22,8 @@ public void Configuration(IAppBuilder app)
                         OnValidateClientAuthentication = (ctx) =>
                         {
                             string clientId, clientSecret;
-                            if (ctx.TryGetBasicCredentials(out clientId, out clientSecret) &&
+                            if ((ctx.TryGetBasicCredentials(out clientId, out clientSecret) ||
+                                 ctx.TryGetFormCredentials(out clientId, out clientSecret)) &&
```

Comment:
Can you split this into two if's so bith credential types are tested with different credentials. This way we're sure the correct method is being used.